src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,419 @@
+PARAM_MODEL_OUT(LinearSVM<>, "output_model", "Output for trained "
+    "linear svm model.", "M");
+
+// Testing.
+PARAM_MATRIX_IN("test", "Matrix containing test dataset.", "T");
+PARAM_UROW_IN("test_labels", "Matrix containing test labels.", "A");
+PARAM_UROW_OUT("predictions", "If test data is specified, this matrix is where "
+    "the predictions for the test set will be saved.", "P");
+PARAM_MATRIX_OUT("score", "If test data is specified, this "
+    "matrix is where the class score for the test set will be saved.",
+    "p");
+
+static void mlpackMain()
+{
+  if (CLI::GetParam<int>("seed") != 0)
+    math::RandomSeed((size_t) CLI::GetParam<int>("seed"));
+  else
+    math::RandomSeed((size_t) std::time(NULL));
+
+  // Collect command-line options.
+  const double lambda = CLI::GetParam<double>("lambda");
+  const double delta = CLI::GetParam<double>("delta");
+  const string optimizerType = CLI::GetParam<string>("optimizer");
+  const double tolerance = CLI::GetParam<double>("tolerance");
+  const bool intercept = CLI::HasParam("no_intercept");
+  const size_t epochs = (size_t) CLI::GetParam<int>("epochs");
+  const size_t maxIterations = (size_t) CLI::GetParam<int>("max_iterations");
+
+  // One of training and input_model must be specified.
+  RequireAtLeastOnePassed({ "training", "input_model" }, true);
+
+  // If no output file is given, the user should know that the model will not be
+  // saved, but only if a model is being trained.
+  RequireAtLeastOnePassed({ "output_model", "predictions", "score"},
+      false, "no output will be saved");
+
+  ReportIgnoredParam({{ "test", false }}, "predictions");
+  ReportIgnoredParam({{ "test", false }}, "score");
+  ReportIgnoredParam({{ "test", false }}, "test_labels");
+
+  // Max Iterations needs to be positive.
+  RequireParamValue<int>("max_iterations", [](int x) { return x >= 0; },
+      true, "max_iterations must be positive or zero");
+
+  // Tolerance needs to be positive.
+  RequireParamValue<double>("tolerance", [](double x) { return x >= 0.0; },
+      true, "tolerance must be positive or zero");
+
+  // Optimizer has to be L-BFGS or parallel SGD.
+  RequireParamInSet<string>("optimizer", { "lbfgs", "psgd" },
+      true, "unknown optimizer");
+
+  // Epochs needs to be non-negative.
+  RequireParamValue<int>("epochs", [](int x) { return x >= 0; }, true,
+      "maximum number of epochs must be non-negative");
+
+  if (optimizerType != "psgd")
+  {
+    if (CLI::HasParam("step_size"))
+    {
+      Log::Warn << PRINT_PARAM_STRING("step_size") << " ignored because "
+          << "optimizer type is not 'psgd'." << std::endl;
+    }
+    if (CLI::HasParam("shuffle"))
+    {
+      Log::Warn << PRINT_PARAM_STRING("shuffle") << " ignored because "
+          << "optimizer type is not 'psgd'." << std::endl;
+    }
+    if (CLI::HasParam("epochs"))
+    {
+      Log::Warn << PRINT_PARAM_STRING("epochs") << " ignored because "
+          << "optimizer type is not 'psgd'." << std::endl;
```

Comment:
Similarly, if `max_iterations` is specified but the optimizer is `psgd`, we should issue a warning.

Reply:
I have added this.